lib/gridfs-stream/upload.js

```diff
@@ -447,3 +486,15 @@ function writeRemnant(_this, callback) {
     checkDone(_this);
   });
 }
+
+/**
+ * @ignore
+ */
+
+function checkAborted(_this, callback) {
```

Comment:
Why not also call this in `writeRemnant`? 
It's called asynchronously from `end()` which does call `checkAborted()`, but only at its beginning, before the asynchronous calls to `waitForIndexes()` or `write()` (in fact the `write()` case is not an issue: internally it calls again `checkAborted()` in `doWrite()`).

So if `abort()` is called after calling `end()`, but before `end()` finishes, then the abort action will be ignored, and the last chunks will be inserted in mongodb, in parallel of `abort()` own `chunks.deleteMany()`: undefined behavior leading to potential chunks leaks.

If I understand correctly it can only happen when `waitForIndexes()` is effectively asynchronous: this is a very narrow case indeed.


To avoid any such issue, no one should touch the db after `abort()` was called, except for index related operations, which are not related to a specific file in gridfs.

For `this.chunks` I believe `writeRemnant()` is the only remaining one missing a `checkAborted()` call.

As for `this.files`, the only one writing in it is `checkDone()`, which could be extra protected for future possible issues, but currently all its callers are or will be protected when `writeRemnant()` will be fixed.